addon/components/infinity-loader.js

```diff
@@ -1,106 +1,65 @@
 import Ember from 'ember';
 import InfinityPromiseArray from 'ember-infinity/lib/infinity-promise-array';
+import InViewportMixin from 'ember-in-viewport';
+import { run } from '@ember/runloop';
+import { computed } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import Component from '@ember/component';
 
-const InfinityLoaderComponent = Ember.Component.extend({
+const InfinityLoaderComponent = Component.extend(InViewportMixin, {
   classNames: ["infinity-loader"],
-  classNameBindings: ["infinityModelContent.reachedInfinity"],
+  classNameBindings: ["infinityModelContent.reachedInfinity", "viewportEntered:in-viewport"],
   guid: null,
   eventDebounce: 10,
   loadMoreAction: 'infinityLoad',
   loadingText: 'Loading Infinite Model...',
   loadedText: 'Infinite Model Entirely Loaded.',
-  destroyOnInfinity: false,
   developmentMode: false,
-  scrollable: null,
   triggerOffset: 0,
 
   willInsertElement() {
     if (this.get('_isInfinityPromiseArray')) {
-      Ember.defineProperty(this, 'infinityModelContent', Ember.computed.alias('infinityModel.content'));
+      Ember.defineProperty(this, 'infinityModelContent', computed.alias('infinityModel.content'));
     } else {
-      Ember.defineProperty(this, 'infinityModelContent', Ember.computed.alias('infinityModel'));
+      Ember.defineProperty(this, 'infinityModelContent', computed.alias('infinityModel'));
     }
   },
 
+  /**
+   * setup ember-in-viewport properties
+   * 
+   * @method didInsertElement
+   */
   didInsertElement() {
     this._super(...arguments);
-    this._setupScrollable();
-    this.set('guid', Ember.guidFor(this));
-    this._bindEvent('scroll');
-    this._bindEvent('resize');
-
-    if (this.get('_isInfinityPromiseArray')) {
-      this.get('infinityModel').then(() => this._loadMoreIfNeeded());
-    } else {
-      this._loadMoreIfNeeded();
-    }
+    this.set('guid', guidFor(this));
```

Comment:
Pretty sure the `guid` property was only used for the scroll event.  Do we need it anymore?